papas/datatypes/src/Cluster.cpp

```diff
@@ -27,19 +27,44 @@ Cluster::Cluster(const Cluster& c, unsigned int index, IdCoder::ItemType type, c
     : m_id(IdCoder::makeId(index, type, subtype, val)),
       m_size(c.m_size),
       m_angularSize(c.m_angularSize),
-      m_pt(c.m_pt),
       m_p3(c.m_p3),
       m_energy(c.m_energy),
       m_subClusters() {
   ;
   m_subClusters.push_back(&c);
 }
 
+Cluster::Cluster(std::list<const Cluster*> overlappingClusters, unsigned int index, char subtype)
+    : m_subClusters(overlappingClusters) {
+  Identifier firstId = -1;
+  char type;
+  for (const auto* cluster : overlappingClusters) {
+    if (cluster->subClusters().size() > 1) {
+      throw "can only merge clusters which are not already merged";
+    }
+    if (firstId == -1) {
+      firstId = cluster->id();
+      m_p3 = cluster->position() * cluster->energy();
+      m_energy = cluster->energy();
+      m_size = cluster->size();  // needed for the case where overlappingClusters has just one cluster
+      m_angularSize = cluster->angularSize();
+    } else {
+      if (IdCoder::typeAndSubtype(firstId) != IdCoder::typeAndSubtype(cluster->id()))
+        throw "Merged Clusters must be made of clusters of same type and subtype";
+      m_p3 += cluster->position() * cluster->energy();  // energy weighted position
```

Comment:
ok, then m_p3 should be called m_position?  
p3 is the name of the momentum vector